File: Recursion/factorial.py
# ans = 1;
# def fibo_num(i):
#     if(i==0):
#         return
#     fibo_num(i-1)
#     global ans 
#     ans = ans * i
def fact(i):
    if(i==0):
        return 1;
    ans = i*fact(i-1)
    return ans

result = fact(5)
print(result)

File: Recursion/factorial.cpp
#include<bits/stdc++.h>
using namespace std;
// int ans = 1; 
// void fibo(int i){
//     if(i==0) return;
//     fibo(i-1);
//     ans = ans*i;
// }

int fact(int i){
    if(i==0) return 1;
    int ans = i* fact(i-1);
    return ans;
}
int main(){
    int result = fact(5); 
   cout<<result;
    return 0;
}

